pkgs/top-level/all-packages.nix

```diff
@@ -15820,11 +15825,22 @@ with pkgs;
 
   default-gcc-version =
     if (with stdenv.targetPlatform; isVc4 || libc == "relibc") then 6
-    else 12;
+    else if stdenv.buildPlatform.isDarwin  then 12  # unable to test
+    else if stdenv.buildPlatform.isAarch64 then 12  # unable to test
+    else 13;
   gcc = pkgs.${"gcc${toString default-gcc-version}"};
   gccFun = callPackage ../development/compilers/gcc;
   gcc-unwrapped = gcc.cc;
 
+  disable-warnings-if-gcc13 = pkg:
+    if (pkg.stdenv.cc.cc.isGNU or false && lib.versionAtLeast pkg.stdenv.cc.cc.version "13.0")
+    then pkg.overrideAttrs(previousAttrs: previousAttrs // {
+      env = previousAttrs.env or {} // {
+        NIX_CFLAGS_COMPILE = (previousAttrs.env.NIX_CFLAGS_COMPILE or "") + " -w";
+      };
+    })
+    else pkg;
+
```

Comment:
Isn't disabling *all* warnings a bit of a sledgehammer?  Can we select just those relevant to gcc 12->13?  (and maybe make them `-Wno-error=foo` so they remain in log)

Reply:
> Can we select just those relevant to gcc 12->13?

I don't know of a guaranteed list of those.

On a more philosophical level, I really wonder what is the point of distributions (rather than upstream developers) enabling compiler warnings.  Does anybody really expect package maintainers (rather than upstream authors) to fix compiler warnings?  Or even *do* anything about them?

This is actually a serious question.  I sort of feel like we're building with `-werror` due to momentum and nobody has ever sat down and given serious thought to *why* we do it.  Since it broke *so much stuff* during this upgrade I think it might be a good time to talk about it.

Reply:
Sure such checks should certainly be stricter upstream than in distros.  So I don't see this comment as a big deal.

For me the rule of thumb is what happens when such a check breaks a build, i.e. if there's someone (around nixpkgs) who's willing to look into it for that particular package (at least report it upstream), or if it will just get broken or the warning disabled.

Reply:
As for the (potential) point of doing checks in distros &ndash; sometimes things in distros turn out differently than upstream anticipated, be it something around dependencies, configure flags, platform, ...

Reply:
> and maybe make them `-Wno-error=foo` so they remain in log

https://github.com/NixOS/nixpkgs/pull/268097/commits/8e38850acc3e268dafccab1b526dbf75ac619179

Reply:
> For me the rule of thumb is what happens when such a check breaks a build, i.e. if there's someone (around nixpkgs) who's willing to look into it for that particular package (at least report it upstream), or if it will just get broken or the warning disabled.

Right, so if the expected result is breakage, followed by manually disabling the warning (or at least making it not-an-error), why don't we just skip the breakage and the manual parts and proactively put `-Wno-error` in `NIX_CFLAGS_COMPILE` by default?  With an explicit `dontAddWnoError` opt-out of course.

I see no downside to this.  It gets us to the same place, but with less breakage and less manual intervention.